src/Symfony/Component/Console/DependencyInjection/AddConsoleCommandPass.php

```diff
@@ -36,21 +50,52 @@ public function process(ContainerBuilder $container)
                 throw new InvalidArgumentException(sprintf('Class "%s" used for service "%s" cannot be found.', $class, $id));
             }
             if (!$r->isSubclassOf(Command::class)) {
-                throw new InvalidArgumentException(sprintf('The service "%s" tagged "console.command" must be a subclass of "%s".', $id, Command::class));
+                throw new InvalidArgumentException(sprintf('The service "%s" tagged "%s" must be a subclass of "%s".', $id, $this->commandTag, Command::class));
             }
 
             $commandId = 'console.command.'.strtolower(str_replace('\\', '_', $class));
-            if ($container->hasAlias($commandId) || isset($serviceIds[$commandId])) {
-                $commandId = $commandId.'_'.$id;
+
+            if (!isset($tags[0]['command'])) {
+                if (isset($serviceIds[$commandId]) || $container->hasAlias($commandId)) {
+                    $commandId = $commandId.'_'.$id;
+                }
+                if (!$definition->isPublic()) {
+                    $container->setAlias($commandId, $id);
+                    $id = $commandId;
+                }
+                $serviceIds[$commandId] = $id;
+
+                continue;
             }
-            if (!$definition->isPublic()) {
-                $container->setAlias($commandId, $id);
-                $id = $commandId;
+
+            $serviceIds[$commandId] = false;
+            $commandName = $tags[0]['command'];
+            $lazyCommandMap[$commandName] = $id;
+            $lazyCommandRefs[$id] = new TypedReference($id, $class);
+            $aliases = array();
+
+            foreach ($tags as $tag) {
+                if (!isset($tag['command'])) {
+                    throw new InvalidArgumentException(sprintf('Missing "command" attribute on tag "%s" for service "%s".', $this->commandTag, $id));
+                }
+                if ($commandName !== $tag['command']) {
+                    throw new InvalidArgumentException(sprintf('The "command" attribute must be the same on each "%s" tag for service "%s".', $this->commandTag, $id));
```

Comment:
Wouldn't it be more straightforward if aliases are just the additional tags using the `command` attribute as well?
Then there is no need for an alias property at all and this strange condition doesn't apply either.

```
tags:
        - { name: console.command, command: app:my-command }
        - { name: console.command, command: app:my-alias }
```